src/dbnode/storage/limits/query_limits.go

```diff
@@ -171,6 +179,36 @@ func (q *queryLimits) AnyExceeded() error {
 	return q.bytesReadLimit.exceeded()
 }
 
+func (q *lookbackLimit) Options() LookbackLimitOptions {
+	return q.options
+}
+
+// Override overrides the limit set on construction.
+func (q *lookbackLimit) Update(opts LookbackLimitOptions) error {
+	if err := opts.validate(); err != nil {
+		return err
+	}
+
+	q.lock.Lock()
+	defer q.lock.Unlock()
+
+	old := q.options
+	q.options = opts
+
+	// If the lookback changed, replace the background goroutine that manages the periodic resetting.
+	if q.options.Lookback != old.Lookback {
+		q.stop()
+		q.start()
```

Comment:
could q.start possibly fail? Scary to imagine if the resetting goroutine doesn't get started, meaning the limits will run amok

Reply:
looks like it shouldn't be able to

Reply:
why don't we change the tick value of the existing goroutine?

Reply:
I had tried to do this but `ticker.Reset` is go 1.15 https://golang.org/doc/go1.15#time

Reply:
just create a new ticker? don't really care, just throwing out ideas

Reply:
Yeah that's what we're doing instead of `.Reset(...)`. We just also call `stop` so the current ticker stops.